src/Microsoft.AspNetCore.Routing.Abstractions/IEndpointMatchConstraint.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace Microsoft.AspNetCore.Routing
+{
+    public interface IEndpointMatchConstraint
+    {
+        void Initialize(string parameter);
+
+        bool Match(string routeKey, RouteValueDictionary values, RouteDirection routeDirection);
```

Comment:
Thoughts on a context object instead of individual parameters? That way something can be added to the context object without breaking interface signature.

There would be 1 context object per route rather than 1 per evaluated constraint to perf shouldn't be impacted too heavily.